app/_how-tos/authenticate-consumers-with-session-and-key-auth.md

```diff
@@ -0,0 +1,182 @@
+entities:
+  consumers:
+    - username: alex
+      keyauth_credentials:
+        - key: hello_world
+    - username: anonymous_users
+      id: 81823632-10c0-4098-a4f7-31062520c1e6
+{% endentity_examples %}
+<!--vale on-->
+
+## 3. Enable the Session plugin
+
+The [Session plugin](/plugins/session/) allows you to manage browser sessions for APIs proxied through the {{site.base_gateway}}.
+
+We'll be setting `config.cookie_secure` to `false` for the sake of this tutorial so we don't have to use HTTPS, but in a production instance, leave this as the default value of `true`.
+
+Enable the plugin for the Gateway Service you created in the [prerequisites](#pre-configured-entities):
+
+<!--vale off-->
+{% entity_examples %}
+entities:
+  plugins:
+    - name: session
+      service: example-service
+      config:
+        storage: kong
+        cookie_secure: false
+{% endentity_examples %}
+<!--vale on-->
+
+## 4. Enable the Request Termination plugin
+
+In this tutorial, we'll be using the [Request Termination plugin](/plugins/request-termination/) to prevent unauthorized access by anonymous Consumers: 
+
+<!--vale off-->
+{% entity_examples %}
+entities:
+  plugins:
+    - name: request-termination
+      service: example-service
+      consumer: anonymous_users
+      config:
+        status_code: 403
+        message: 'Forbidden'
+{% endentity_examples %}
+<!--vale on-->
+
+## 5. Validate
+
+After configuring the Key Authentication Encryption plugin, you can verify that it was configured correctly and is working by sending requests with and without the API key you created for your Consumer.
+
+First, run the following to verify that anonymous requests return an error:
+
+<!--vale off-->
+{% validation unauthorized-check %}
+url: /anything
+status_code: 403
+message: 'Forbidden'
+{% endvalidation %}
+<!--vale on-->
+
+Then, run the following command to verify that a user can authenticate via sessions:
+
+<!--vale off-->
+{% validation unauthorized-check %}
+url: /anything
+headers:
+  - 'apikey: hello_world'
+message: OK
+status_code: 200
+{% endvalidation %}
+<!--vale on-->
+
+The response should now have the `Set-Cookie` header.
+Make sure that this cookie works by copying the contents (for example: `session=emjbJ3MdyDsoDUkqmemFqw..|1544654411|4QMKAE3I-jFSgmvjWApDRmZHMB8.`) and exporting it in an environment variable:
+
+```sh
+export COOKIE_HEADER='<your-cookie-header-here>'
+```
+
+Use your session token in the request:
+<!--vale off-->
+{% validation unauthorized-check %}
+url: /anything
+headers:
+  - 'cookie: "$COOKIE_HEADER"'
+message: OK
+status_code: 200
+{% endvalidation %}
+<!--vale on-->
+
+Now, if you try to authenticate without sending a Key Auth credential, you will still be authenticated because {{site.base_gateway}} is using the session cookie that is granted by the Session plugin:
+
+<!--vale off-->
+{% validation request-check %}
+url: /anything
+status_code: 200
+message: 'OK'
+{% endvalidation %}
+<!--vale on-->
```

Comment:
```suggestion
```
This doesn't really make sense as a step - the credentials are stored in the cookie now, so you definitely need to pass the cookie to be authenticated. So you can just ditch this portion.